packages/mgt-components/README.md

```diff
@@ -55,6 +55,40 @@ The components can be used on their own, but they are at their best when they ar
     ```
 
+
+## <a id="disambiguation">Disambiguation</a>
+
+MGT is built using [web components](https://developer.mozilla.org/en-US/docs/Web/Web_Components). Web components use their tag name as a unique key when registering within a browser. Any attempt to register a component using a previously registered tag name results in an error being thrown when calling `CustomElementRegistry.define()`. In scenarios where multiple custom applications can be loaded into a single page this created issues for MGT, most notably in developing custom SharePoint web parts.
+
+To mitigate this challenge the [`mgt-spfx`](https://github.com/microsoftgraph/microsoft-graph-toolkit/tree/main/packages/mgt-spfx) package was built. This provided a library extension to SharePoint which centralized the registration of the MGT web components. This allowed for multiple MGT web parts to be loaded into a single page without throwing errors. When using `mgt-spfx` all MGT based web parts in a SharePoint tenant used the same version of MGT.
+
+In order to allow developers to build custom MGT web parts using the latest version of MGT and retain the ability to have web parts built using v2.x of MGT on the same page a new disambiguation feature has been added to MGT. This allows developers to specify a unique string to add to the tag name of all MGT web components in their application.
+
+The earlier example can be updated to use the disambiguation feature as follows:
+
+```html
+<script type="module">
+  import { Providers, customElementHelper } from '@microsoft/mgt-element';
+  customElementHelper.withDisambiguation('contoso');
+  import { Msal2Provider } from '@microsoft/mgt-msal2-provider';
+  Providers.globalProvider = new Msal2Provider({clientId: 'clientId'});
+
+</script>
+
+<mgt-contoso-login></mgt-contoso-login>
+<mgt-contoso-person person-query="Bill Gates" person-card="hover"></mgt-contoso-person>
+<mgt-contoso-agenda group-by-day></mgt-contoso-agenda>
+```
+> Note: the `import` of `mgt-components` must use a dynamic import to ensure that the disambiguation is applied before the components are imported. Helper utilities have been provided in the  [`mgt-spfx-utils`](https://github.com/microsoftgraph/microsoft-graph-toolkit/tree/main/packages/mgt-spfx-utils) package to make this easier in the context of SharePoint web parts.
+
+### Dynamic imports aka Lazy Loading
+
+Dynamic imports are a means to asynchronously load code. This is useful when you want to load code only when it is needed. For example, you may want to load a component only when a user clicks a button. This is a great way to reduce the initial load time of your application. In the context of disambiguation it is necessary to use this technique as the components register themselves in the browser when they are imported.
+
+When using an `import` statement the import statement is hoisted and executed before any other code in the code block. To use dynamic imports you must use the `import()` function.
+
```

Comment:
Let's add a code sample where we show static import and its equivalent dynamic import, to help developers understand what we're talking about. Perhaps they already know the concept, but don't know that it's called _dynamic imports_ so a code sample could help them put the two together.

Reply:
Sample added and re-ordered the section slightly

Reply:
Just to make it even clearer, shall we add some surrounding code to make it explicit that a static import goes at the top of the file, while a dynamic import is located somewhere in the code, where you want to load it explicitly? I think that would help folks who've never seen the pattern understand it better.

Reply:
Yeah that's a good idea. Although the import statement at the top of the file is a good practice due to the fact it gets hoisted, not a language requirement.

Reply:
@waldekmastykarz I've expanded this using our continued example